<launch>
<arg name="play_bag" default="true"/>
<param name="/use_sim_time" value="true" />

<!-- <node pkg="tf" type="static_transform_publisher" name="link1_broadcaster" args="0 0 0 0 0 0 1 map odom 100" />
<node pkg="submap" name="Submap_building" type="Submap_building" output="screen" required="true">
        <remap from="/points" to="/sampled_points" />
</node> -->

<!-- <node pkg="submap" name="rviz_gmm" type="basic_shapes" output="screen" required="true"/> -->
<!-- <node pkg="gmm_map_python" name="talker" type="gmm_transform.py" output="screen" required="true"/> -->
<!-- <node pkg="gmm_map_python" name="visualgmm" type="visualgmm.py" output="screen" required="true"/> -->

<group ns="robot2">
<arg name="robot_name" default="robot2"/>

<node pkg="gmm_map_python" name="Downsample" type="DownSampleNode" output="log" required="true">
        <remap from="Downsample/points" to="d400/depth/color/points"/>
        <remap from="Downsample/sampled_points" to="Downsample/sampled_points"/>
</node>
<node pkg="gmm_map_python" name="MapBuilderNode" type="MapBuilderNode.py" output="screen" required="true">
        <remap from="sampled_points" to="Downsample/sampled_points"/>
        <param name="baselink_frame" value="$(arg robot_name)/t265_link"/>
        <param name="odom_frame" value="$(arg robot_name)/t265_odom_frame"/>
        <param name="map_frame" value="$(arg robot_name)/map"/>
        <param name="robot_id" value="2"/>
</node>

<node pkg="rviz" name="rvizview" type="rviz" output="screen" args="-d $(find gmm_map_python)/rviz/$(arg robot_name).rviz" required="false"/>

</group>

<node if="$(arg play_bag)" pkg="rosbag" type="play" name="recordplay_2" args="--clock /home/xuzhl/dataset/realsence_t265/output_robot2_notime.bag" required="false">
</node> 

</launch>
